sklearn/naive_bayes.py

```diff
@@ -117,6 +118,303 @@ def predict_proba(self, X):
         return np.exp(self.predict_log_proba(X))
 
 
+class GeneralNB(_BaseNB, _BaseComposition, ClassifierMixin):
+    """General Naive Bayes
+
```

Comment:
A brief description would do well here, e.g. "Allows the user to compose a joint probability distribution as the product of independent models or probability distributions. Each constituent distribution is fitted on a subset of features."

Reply:
Updated.